### PR TITLE
Made empty string checking more idomatic. This fixes #13

### LINKED ISSUE #13: Change all checks for empty strings
Change checks fro empty strings from:

```
someString := ""

if len(someString) == 0 {
    ...
}
```
to 

```
someString := ""

if someString == "" {
    ...
}
```

to have a more [idiomatic go](https://dmitri.shuralyov.com/idiomatic-go#empty-string-check).

### PR DESCRIPTION
**Pull request by WerWolv**

### CODE REVIEW

File: pkg/aws/rekognition/util.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func unmarshalRekognitionOut(input, output interface{}) error {
 
 	m := structs.Map(input)
 
-	if len(m) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: This is a map, not a string.

File: pkg/aws/rekognition/util.go [hunk 2]
```diff
@@ -193,11 +193,11 @@ func unmarshalRekognitionOut(input, output interface{}) error {
 // NewCompareFacesInput builds a *rekognition.CompareFacesInput starting from the two images, their bucket and a similarity threshold
 func NewCompareFacesInput(source, target []byte, similarity float64) (*rekognition.CompareFacesInput, error) {
 
-	if len(source) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: These are bytes, not strings.

File: pkg/aws/rekognition/util.go [hunk 3]
```diff
@@ -193,11 +193,11 @@ func unmarshalRekognitionOut(input, output interface{}) error {
...
+	if source == "" {
 		return nil, intErr.Format(Source, ErrEmptyParameter)
 	}
 
-	if len(target) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: These are bytes, not strings.

File: pkg/aws/rekognition/util.go [hunk 4]
```diff
@@ -226,7 +226,7 @@ func NewCompareFacesInput(source, target []byte, similarity float64) (*rekogniti
 // NewDetectFacesInput builds a *rekognition.DetectFacesInput starting from the image
 func NewDetectFacesInput(source []byte) (*rekognition.DetectFacesInput, error) {
 
-	if len(source) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: These are bytes, not strings.

File: pkg/aws/rekognition/util.go [hunk 5]
```diff
@@ -245,7 +245,7 @@ func NewDetectFacesInput(source []byte) (*rekognition.DetectFacesInput, error) {
 // NewDetectTextInput builds a *rekognition.DetectTextInput starting from the image
 func NewDetectTextInput(source []byte) (*rekognition.DetectTextInput, error) {
 
-	if len(source) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: These are bytes, not strings.

File: pkg/aws/rekognition/util.go [hunk 6]
```diff
@@ -264,7 +264,7 @@ func NewDetectTextInput(source []byte) (*rekognition.DetectTextInput, error) {
 // newInputImage returns a *rekognition.Image given an S3 image []byte encoded
 func newInputImage(image []byte) (*rekognition.Image, error) {
 
-	if len(image) == 0 {
```
Comment (andream16) [inline pkg/aws/rekognition/util.go]: These are bytes, not strings.

File: pkg/aws/s3/util.go [hunk 7]
```diff
@@ -51,7 +51,7 @@ func UnmarshalGetObjectOutput(input *s3.GetObjectOutput) ([]byte, error) {
 	if bytesErr != nil {
 		return nil, bytesErr
 	}
-	if len(body) == 0 {
```
Comment (andream16) [inline pkg/aws/s3/util.go]: These are bytes, not strings.

File: pkg/aws/s3/util.go [hunk 8]
```diff
@@ -135,16 +135,16 @@ func NewGetObjectInput(bucketName, source string) (*s3.GetObjectInput, error) {
...
 	}
-	if len(fileName) == 0 {
+	if fileName == "" {
 		return nil, intErr.Format(FileName, ErrEmptyParameter)
 	}
-	if len(contentType) == 0 {
+	if contentType == "" {
 		return nil, intErr.Format(ContentType, ErrEmptyParameter)
 	}
-	if len(image) == 0 {
```
Comment (andream16) [inline pkg/aws/s3/util.go]: These are bytes, not strings.

**Comment by WerWolv** [comment]
Should maybe have checked better. Sorry for that

**Review by andream16 — CHANGES_REQUESTED** [review]
Just continue checking length on maps, slices and bytes. Good job for strings.

File: pkg/aws/dynamodb/util.go [hunk 9]
```diff
@@ -75,7 +75,7 @@ func UnmarshalStreamImage(input events.DynamoDBEventRecord, output interface{})
 		return intError.Format(ErrNoPointerParameter, Output)
 	}
 
-	if len(img) == 0 {
```
Comment (andream16) [inline pkg/aws/dynamodb/util.go]: This is a map.

**Review by andream16 — CHANGES_REQUESTED** [review]
Fix map check.

**Review by andream16 — APPROVED** [review]
Good job.

**Comment by WerWolv** [comment]
I'm really really sorry for all that struggle. Hope it still helped a little

**Pull request merged**